### ISSUE TITLE
Déplacer l'initialisation du type de provider dans les interfaces correspondantes

### ISSUE THREAD

**Issue by mxmeunier** [betagouv/api-subventions-asso#378]
Pourquoi isEtablissementProvider est défini dans l'implémentation et pas directement dans l'interface EtablissementProvider ?

Tous les champs devraient être à false dans I'interface Provider et les sous implémentation (Association, Etablissement) devraient mettre à jour leurs attributs.

**Comment by VSerain** [comment]
Je rappelle juste qu'une interface ne peut pas set des attribues, elle ne serve qu'à décrire leurs types.

Si j'ai bien compris ton ticket, il faudrait les transformer en class pour pouvoir en étendre.

Pour ce qui est de mettre les autres à false, je dirais que c'est une question de point de vu, en JS undefined est falsy, c'est comme si elle etait déjà à false. 
D'un point de vue exécution, je n'en vois pas l'intérêt.
D'un point de vue propreté du code, why not, mais il faut prendre en compte que le js n'a pas de fonctionnement d'héritage multiples !

**Comment by mxmeunier** [comment]
Effectivement je n'avais pas vu qu'on parlait d'interface et pas de classe.

Le problème du ticket c'est qu'on a à chaque fois isEtablissementProvider = true alors qu'au dessus on a entends EtablissementProvider... Donc j'aimerai trouver un moyen d'avoir le isEtablissementProvider à true. En gros l'idée serait peut-être de transformer EtablissementProvider en classe qui étend l'interface Provider qui aurait tous les champs provider... À voir !

**Comment by VSerain** [comment]
Oui, mais ça pause un souci d'héritages multiples, on pourra se faire un call à voix haute, ça sera plus simple pour trouver une idée !

**Comment by VSerain** [comment]
@mxmeunier Il faut qu'on ce call un point la dessus

**Comment by mxmeunier** [comment]
@VSerain avant ton départ en vacances ?
